AssemblerLib/LocalDataInitializer.h

```diff
@@ -240,18 +242,19 @@ class LocalDataInitializer
 
     /// Sets the provided data_ptr to the newly created local assembler data.
     void operator()(
-            const MeshLib::Element& e,
+            std::size_t const mesh_item_id,
+            MeshLib::Element const& element,
             LADataIntfPtr& data_ptr,
-            std::size_t const local_matrix_size,
             unsigned const integration_order,
-            ConstructorArgs&&... args)
+            ConstructorArgs&&... args) const
     {
-        auto const type_idx = std::type_index(typeid(e));
-        auto it = _builder.find(type_idx);
+        auto const type_idx = std::type_index(typeid(element));
+        auto const it = _builder.find(type_idx);
 
         if (it != _builder.end()) {
+            auto const num_local_dof = _dof_table.getNumElementDOF(mesh_item_id);
```

Comment:
It looks `_dof_table` is used only here. If the object is needed only for having the number of local DoF, you can pass the number in the constructor.